modules/management-console-agent/src/main/java/org/apache/ignite/internal/agent/action/controller/QueryActionsController.java

```diff
@@ -89,6 +89,27 @@ public QueryActionsController(GridKernalContext ctx) {
         return new IgniteFinishedFutureImpl<>();
     }
 
+    /**
+     * @param globalQryId Global query id.
+     * @return Kill query result.
+     */
+    public IgniteInternalFuture<QueryResult> kill(String globalQryId) {
+        return ctx.closure().callLocalSafe(() -> {
+            SqlFieldsQuery qryFields = new SqlFieldsQuery("KILL QUERY '" + globalQryId + "';");
+
+            try (
+                FieldsQueryCursor<List<?>> cur = qryProc.querySqlFields(qryFields, true);
+                CursorHolder cursorHolder = new CursorHolder(cur)
+            ) {
+                QueryResult res = fetchSqlQueryResult(cursorHolder, 1);
+
+                res.setResultNodeId(ctx.localNodeId().toString());
+
+                return res;
```

Comment:
Is the result going to be used the frontend?

Reply:
I think no, make it as cancel action :)